src/action.py

```diff
@@ -140,39 +138,55 @@ def _update(
         raise exceptions.ActionError(
             f"internal error, content change for page is None, {action=!r}"
         )
-    if (
-        action.navlink_change.new.link is not None
-        and action.content_change is not None
-        and action.content_change.new is None
-    ):
-        raise exceptions.ActionError(f"internal error, new content for page is None, {action=!r}")
 
     if (
         action.content_change is not None
-        and action.content_change.old is not None
-        and action.content_change.new is not None
+        and action.content_change.server is not None
+        and action.content_change.local is not None
     ):
-        _log_content_change(old=action.content_change.old, new=action.content_change.new)
+        _log_content_change(
+            base=action.content_change.base or action.content_change.server,
+            new=action.content_change.local,
+        )
 
-    if (
+    # All of these are needed to ensure types are as expected
+    if (  # pylint: disable=too-many-boolean-expressions
         not dry_run
         and action.navlink_change.new.link is not None
         and action.content_change is not None
-        and action.content_change.new is not None
-        and action.content_change.new != action.content_change.old
+        and action.content_change.base is not None
+        and action.content_change.local is not None
+        and action.content_change.local != action.content_change.server
```

Comment:
Perhaps I'm missing something but, what if `action.content_change.server` is None?

Reply:
Actually, you helped identify some unnecessary checks. Both `action.content_change.server` and `action.content_change.local` can no longer be None, it used to be possible but that was changed in this PR

Reply:
should be fixed in the next push